app/src/main/java/com/example/tasky/feature_agenda/presentation/AgendaViewModel.kt

```diff
@@ -59,9 +68,24 @@ class AgendaViewModel @Inject constructor(
         }
     }
 
-    fun getInitials(): String {
-        // todo grab user name stored in shared pref and parse firstName.first() lastName.first()
-        return "AB"
+    fun getInitials() {
+        viewModelScope.launch {
+            val fullName = taskyState.getName()
+            val nameSplit = fullName.split(" ")
+            var initials = ""
+
+            if (nameSplit.size == 1) {
+                val name = nameSplit.first()
+                initials = "${name[0]}${name[1]}"
+            } else {
+                for (i in nameSplit.indices) {
+                    if (i == 0 || i == nameSplit.lastIndex) {
+                        initials += nameSplit[i].first()
+                    }
+                }
+            }
+            _initials.value = initials.uppercase()
+        }
```

Comment:
Could it make sense extracting that logic into a separate function independent of the ViewModel? Then you could reuse it for the agenda detail screen as well (attendee section). 
Also, your code will break for a name with leading, trailing or double spaces in between.